yara-x/src/modules/protos/macho.proto

```diff
@@ -0,0 +1,227 @@
+syntax = "proto2";
+
+import "yara.proto";
+
+option (yara.module_options) = {
+  name : "macho"
+  root_message: "Macho"
+  rust_module: "macho"
+};
+
+message Section {
+  optional string segname = 1;
+  optional string sectname = 2;
+  optional uint64 addr = 3;
+  optional uint64 size = 4;
+  optional uint32 offset = 5;
+  optional uint32 align = 6;
+  optional uint32 reloff = 7;
+  optional uint32 nreloc = 8;
+  optional uint32 flags = 9;
+  optional uint32 reserved1 = 10;
+  optional uint32 reserved2 = 11;
+  optional uint32 reserved3 = 12;
+}
+
+message Segment {
+  optional uint32 cmd = 1;
+  optional uint32 cmdsize = 2;
+  optional string segname = 3;
+  optional uint64 vmaddr = 4;
+  optional uint64 vmsize = 5;
+  optional uint64 fileoff = 6;
+  optional uint64 filesize = 7;
+  optional uint32 maxprot = 8;
+  optional uint32 initprot = 9;
+  optional uint32 nsects = 10;
+  optional uint32 flags = 11;
+  repeated Section sections = 12;
+}
+
+message Macho {
+  // Set Mach-O header and basic fields
+  optional uint32 magic = 1;
+  optional uint32 cputype = 2;
+  optional uint32 cpusubtype = 3;
+  optional uint32 filetype = 4;
+  optional uint32 ncmds = 5;
+  optional uint32 sizeofcmds = 6;
+  optional uint32 flags = 7;
+  optional uint32 reserved = 8;
+  optional uint64 number_of_segments = 9;
+  repeated Segment segments = 10;
+  optional uint64 entry_point = 11;
+  optional uint64 stack_size = 12;
+
+  // Set Mach-O definitions
+  required uint32 MH_MAGIC = 13;
```

Comment:
Thank you! This looks great and I will implement it soon. Thank you also for adding this into documentation, at first I didn't connect it (enums) with my usecase but now it is much clearer.

Reply:
While working on this I have ran into an issue, that has been also previously mentioned on keybase. For enums in protobuf there is currently support for numbers up to `0x7fffffff`. This does not allow me to implement enums for multiple fields such a:

- `MH_MAGIC`, `MH_CIGAM`, ... , `S_ATTR_PURE_INSTRUCTIONS` , etc.

Is there any recommended way how to (for now at least) handle those values? Thank you

Reply:
I've added a new feature for supporting larger enum items. See https://github.com/VirusTotal/yara-x/commit/7539bb8861dd4308fb2887352595a560937307f3 and ...

https://github.com/VirusTotal/yara-x/blob/7539bb8861dd4308fb2887352595a560937307f3/docs/Module%20Developer's%20Guide.md?plain=1#L808-L823

Reply:
Thank you for this, I have commited a change that should resolve this. Although I have observed two things during this change:

1. You cannot have same values in one enum (this makes completely sense)
So for example: 
```
enum CPU_INTEL_SUBTYPE {
  option (yara.enum_options).inline = true;
  CPU_SUBTYPE_INTEL_MODEL_ALL = 0x00000000;
  CPU_SUBTYPE_386 = 0x00000003;
  CPU_SUBTYPE_486 = 0x00000004;
  CPU_SUBTYPE_486SX = 0x00000084;
  CPU_SUBTYPE_586 = 0x00000005;
  CPU_SUBTYPE_CELERON = 0x00000067;
  CPU_SUBTYPE_CELERON_MOBILE = 0x00000077;
  CPU_SUBTYPE_ITANIUM = 0x0000000b;
  CPU_SUBTYPE_ITANIUM_2 = 0x0000001b;
  CPU_SUBTYPE_XEON = 0x0000000c;
  CPU_SUBTYPE_XEON_MP = 0x0000001c;
}

enum CPU_I386_SUBTYPE {
  option (yara.enum_options).inline = true;
  CPU_SUBTYPE_I386_ALL = 0x00000003;
}

enum CPU_X86_SUBTYPE {
  option (yara.enum_options).inline = true;
  CPU_SUBTYPE_X86_64_ALL = 0x00000003;
}
```
  intel subtypes had to be split like this.

2. You have to access some of those values (e.g. `MH_MAGIC`) from Rust code. This can be done by `HEADER::MH_MAGIC`, but this only works for enums that does't use `[(yara.enum_value).i64 = 0xXXXXXXX]` as it only access enum value and not the descriptor value. So for example:

      `HEADER::MH_MAGIC` with definition
       `MH_MAGIC = 0 [(yara.enum_value).i64 = 0xfeedface];` 
       returns `0`, 
 
      `HEADER::MH_CIGAM` with definition
      `MH_CIGAM = 1 [(yara.enum_value).i64 = 0xcefaedfe];` 
      returns `1`
      etc,...
      
      So I have decided to also keep definitions for constants that are used in the code at the top of the file.